### ISSUE TITLE
docker-build-cacher-1.9.1 build failure with language-docker-3

### ISSUE THREAD

**Issue by DanBurton** [seatgeek/docker-build-cacher#1]
(I'm not quite sure how the Stackage nightly build gave this a pass; it should have caught the error.)

Repro:

```bash
stack unpack docker-build-cacher-1.9.1 && cd docker-build-cacher-1.9.1
echo 'resolver: nightly-2018-03-12' > stack.yaml
stack build
```

error:

```
[1 of 2] Compiling Main             ( app/Main.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/docker-build-cacher/docker-build-cacher-tmp/Main.o )

/Users/dan/scratch/docker-build-cacher-1.9.1/app/Main.hs:367:24: error:
    • Couldn't match type ‘[Char]’ with ‘Image’
      Expected type: Image
        Actual type: String
    • In the first argument of ‘tagged’, namely ‘Text.unpack t’
      In the second argument of ‘($)’, namely
        ‘Text.unpack t `tagged` "latest"’
      In a stmt of a 'do' block: from $ Text.unpack t `tagged` "latest"
    |
367 |                 from $ Text.unpack t `tagged` "latest" -- Use the cached image as base for the new one
    |                        ^^^^^^^^^^^^^

/Users/dan/scratch/docker-build-cacher-1.9.1/app/Main.hs:400:31: error:
    • Couldn't match type ‘[Char]’ with ‘Image’
      Expected type: Image
        Actual type: String
    • In the first argument of ‘tagged’, namely ‘(Text.unpack tag)’
      In the first argument of ‘from’, namely
        ‘(tagged (Text.unpack tag) "latest")’
      In a stmt of a 'do' block: from (tagged (Text.unpack tag) "latest")
    |
400 |                 from (tagged (Text.unpack tag) "latest")
    |                               ^^^^^^^^^^^^^^^

/Users/dan/scratch/docker-build-cacher-1.9.1/app/Main.hs:479:25: error:
    • Couldn't match type ‘Image’ with ‘[Char]’
      Expected type: String
        Actual type: Image
    • In the first argument of ‘Text.pack’, namely ‘name’
      In the expression: Text.pack name
      In the first argument of ‘Just’, namely
        ‘(Text.pack name, pos, Text.pack alias)’
    |
479 |         Just (Text.pack name, pos, Text.pack alias)
    |                         ^^^^

/Users/dan/scratch/docker-build-cacher-1.9.1/app/Main.hs:481:25: error:
    • Couldn't match type ‘Image’ with ‘[Char]’
      Expected type: String
        Actual type: Image
    • In the first argument of ‘Text.pack’, namely ‘name’
      In the expression: Text.pack name
      In the first argument of ‘Just’, namely
        ‘(Text.pack name, pos, Text.pack alias)’
    |
481 |         Just (Text.pack name, pos, Text.pack alias)
    |                         ^^^^

/Users/dan/scratch/docker-build-cacher-1.9.1/app/Main.hs:483:25: error:
    • Couldn't match type ‘Image’ with ‘[Char]’
      Expected type: String
        Actual type: Image
    • In the first argument of ‘Text.pack’, namely ‘name’
      In the expression: Text.pack name
      In the first argument of ‘Just’, namely
        ‘(Text.pack name, pos, Text.pack alias)’
    |
483 |         Just (Text.pack name, pos, Text.pack alias)
    |                         ^^^^

/Users/dan/scratch/docker-build-cacher-1.9.1/app/Main.hs:524:39: error:
    • Couldn't match type ‘[Char]’ with ‘Image’
      Expected type: Image
resolver: nightly-2018-03-12
        Actual type: String
    • In the first argument of ‘TaggedImage’, namely ‘(Text.unpack t)’
      In the first argument of ‘From’, namely
        ‘(TaggedImage (Text.unpack t) "latest" (formatAlias stageAlias))’
      In the expression:
        From
          (TaggedImage (Text.unpack t) "latest" (formatAlias stageAlias))
    |
524 |                 in From (TaggedImage (Text.unpack t) "latest" (formatAlias stageAlias))
    |                                       ^^^^^^^^^^^^^
```

**Comment by lorenzo** [comment]
@DanBurton This is because of the unbounded version for `language-docker` that just released a new breaking version (3.0)

**Comment by lorenzo** [comment]
Released a new version that fixes this issue. Thanks for reporting!
